src/qibo/models.py

```diff
@@ -429,3 +431,161 @@ def _loss(params, qaoa, hamiltonian):
                                                              compile=compile, processes=processes)
         self.set_parameters(parameters)
         return result, parameters, extra
+
+
+class Grover:
+    """Model that performs Grover's algorithm.
+
+    # TODO: Add a few more details on Grover's algorithm and/or reference.
+
+    Args:
+        oracle (:class:`qibo.core.circuit.Circuit`): quantum circuit that flips
+            the sign using a Grover ancilla initialized with -X-H-
+            and expected to have the total size of the circuit.
+        initial_state (:class:`qibo.core.circuit.Circuit`): quantum circuit
+            that initializes the state. Leave empty if |000..00>
+        superposition (:class:`qibo.core.circuit.Circuit`): quantum circuit that
+            takes an initial state to a superposition. Expected to use the first
+            set of qubits to store the relevant superposition.
+        sup_qubits (int): number of qubits that store the relevant superposition.
+            Leave empty if superposition does not use ancillas.
+        sup_size (int): how many states are in a superposition.
+            Leave empty if its an equal superposition of quantum states.
+        num_sol (int): number of expected solutions. Needed for normal Grover.
```

Comment:
I would suggest to name these variables with the full prefix name, so it is easier for interpretation.